ganttproject/src/net/sourceforge/ganttproject/gui/ResourceAssignmentsTableModel.java

```diff
@@ -0,0 +1,179 @@
+package net.sourceforge.ganttproject.gui;
+
+import net.sourceforge.ganttproject.language.GanttLanguage;
+import net.sourceforge.ganttproject.resource.HumanResource;
+import net.sourceforge.ganttproject.task.ResourceAssignment;
+import net.sourceforge.ganttproject.task.ResourceAssignmentMutator;
+import net.sourceforge.ganttproject.task.Task;
+
+import javax.swing.table.AbstractTableModel;
+import java.util.*;
+
+/**
+ * @author Oleg Kushnikov
+ */
+public class ResourceAssignmentsTableModel extends AbstractTableModel {
+  enum Column {
+    ID("id", String.class),
+    NAME("taskname", String.class),
+    UNIT("unit", String.class);
+
+    private final String myCaption;
+    private final Class<?> myClass;
+
+    Column(String key, Class clazz) {
+      myCaption = GanttLanguage.getInstance().getText(key);
+      myClass = clazz;
+    }
+
+    String getCaption() {
+      return myCaption;
+    }
+
+    public Class<?> getColumnClass() {
+      return myClass;
+    }
+  }
+
+  private final List<ResourceAssignment> myAssignments;
+  private final List<ResourceAssignment> myAssignmentsToDelete = new ArrayList<>();
+  private final HumanResource myResource;
+  private final Map<Task, ResourceAssignmentMutator> myTask2MutatorMap = new HashMap<>();
+
+
+  public ResourceAssignmentsTableModel(HumanResource person) {
+    myResource = person;
+    myAssignments = new ArrayList<>(Arrays.asList(person.getAssignments()));
+  }
+
+  @Override
+  public Class<?> getColumnClass(int columnIndex) {
+    return Column.values()[columnIndex].getColumnClass();
+  }
+
+  @Override
+  public int getRowCount() {
+    return myAssignments.size() + 1;
+  }
+
+  @Override
+  public boolean isCellEditable(int row, int col) {
+    if (row == myAssignments.size()) {
+      return true;
```

Comment:
in the task description you shared with me originally this field was required to be editable. But I agree that if we consider the primary goal o this page to be high-level mapping of the resource to tasks than add/delete  should be enough of edit possibilities